docs/adrs/adr-002-deps-mngmt.md

```diff
@@ -0,0 +1,17 @@
+## Context
+
+Ideally all developers working on a project should never fall into the pitfall of “it works on my machine” - the tools they are using, its configuration should be identical all across the systems, resulting in reproducible artifacts and behaviours, from the dev hosts up to “production environments”. 
+Achieving said goal is not a trivial task however - retaining reproducibility has been traditionally an utopia, where concessions were frequently made due to practicalities of maintaining identical setup across many different machines. State of affairs chiefly influenced by a myriad of system-wide dependencies in many tools, which make things more difficult.
+We would like to provision working environments to developers, users and production-environments that are reproducible in their nature, across all setup, where everyone is using exactly the same binaries, dependencies and configuration. 
+
+## Decision
+We will use [nix package manager](https://nixos.org/) as the vehicle for obtaining reproducibility and hermeticity (hard to have one without other) of our project working environments, as well as delivering hard-pinned dependencies of components traditionally delivered from the OS-level. 
```

Comment:
"as the means of" vs "as the vehicle for" - just sounds better to me.

Reply:
Fixed, thanks!